demos/python_demos/rag_chatbot/servable_stream/model.py

```diff
@@ -0,0 +1,278 @@
+        loader = loader_class(file_path, **loader_args)
+        return loader.load()
+
+    raise ValueError(f"File does not exist '{ext}'")
+
+
+def default_partial_text_processor(partial_text: str, new_text: str):
+    """
+    helper for updating partially generated answer, used by default
+
+    Params:
+      partial_text: text buffer for storing previosly generated text
+      new_text: text update for the current step
+    Returns:
+      updated text string
+
+    """
+    partial_text += new_text
+    return partial_text
+
+
+text_processor = llm_model_configuration.get(
+    "partial_text_processor", default_partial_text_processor
+)
+
+
+def deserialize_prompts(batch_size, input_tensor):
+    if batch_size == 1:
+        return [bytes(input_tensor).decode()]
+    np_arr = deserialize_bytes_tensor(bytes(input_tensor))
+    return [arr.decode() for arr in np_arr]
+
+
+def serialize_completions(batch_size, result):
+    if batch_size == 1:
+        return [Tensor("completion", result.encode())]
+    return [Tensor("completion", serialize_byte_tensor(
+        np.array(result, dtype=np.object_)).item())]
+
+
+
+class OvmsPythonModel:
+    def initialize(self, kwargs: dict):
+        print("Loading LLM model...")
+        self.ov_model = model_class.from_pretrained(
+            llm_model_dir,
+            device="CPU",
+            ov_config=ov_config,
+            compile=True,
+            config=AutoConfig.from_pretrained(llm_model_dir, trust_remote_code=True),
+            trust_remote_code=True)
+        print("LLM model loaded")
+        print("Loading embedding model...")
+        self.embedding = OVEmbeddings.from_model_id(
+            embedding_model_dir,
+            do_norm=embedding_model_configuration["do_norm"],
+            ov_config={
+                "device_name": "CPU",
+                "config": {"PERFORMANCE_HINT": "THROUGHPUT"},
+            },
+            model_kwargs={
+                "model_max_length": 512,
+            },
+        )
+        print("Embedding model loaded")
+        print("Building document database...")
+
+        documents = []
+        for file_path in os.listdir("/documents"):
+            abs_path = f"/documents/{file_path}"
+            print(f"Reading document {abs_path}...")
+            documents.extend(load_single_document(abs_path))
+
+        spliter_name = "RecursiveCharacter"  # TODO: Param?
+        chunk_size=1000  # TODO: Param?
+        chunk_overlap=200  # TODO: Param?
+        text_splitter = TEXT_SPLITERS[spliter_name](chunk_size=chunk_size, chunk_overlap=chunk_overlap)
+        self.texts = text_splitter.split_documents(documents)
+        self.db = Chroma.from_documents(self.texts, self.embedding)
+        vector_search_top_k = 6  # TODO: Param?
+        self.retriever = self.db.as_retriever(search_kwargs={"k": vector_search_top_k})
+
+        print("Document database loaded")
+
+    def execute(self, inputs: list):
+        print("Exec")
```

Comment:
Please remove it or make it more meaningful

Reply:
Done